index.js

```diff
@@ -17,7 +19,13 @@ module.exports = function (str, filepath) {
 
 	return pify(mkdirp, Promise)(path.dirname(fullpath))
 		.then(function () {
-			return pify(fs.writeFile, Promise)(fullpath, str);
+			return isStream(str) ?
+				new Promise((resolve, reject) => {
+					str.pipe(fs.createWriteStream(fullpath))
```

Comment:
> I'm always confused by streams.

Me too. [whatwg/streams](https://github.com/whatwg/streams) can't come soon enough.

As for errors on `str`, I don't really know what is expected here. Unhandled EventEmitter errors get raised as exceptions and crash the process. If they are trapped for any reason, would my writable stream hang on to its file descriptor forever? Seems like it might. I guess we should forward errors to the writable stream (as opposed to just rejecting immediately) so that the writable stream can clean up. Not sure what the precise consequences are, though.

If we were to handle errors on behalf of the user, I think it would look like this:

```js
const pipeStream = () => {
    return new Promise((resolve, reject) => {
        var writeable = fs.createWriteStream(fullpath);
        str
            .on('error', function (err) {
                writeable.emit('error', err);
            })
            .pipe(writeable)
            .on('error', reject)
            .on('finish', resolve);
    });
};
```